src/proxy/http/dialect.rs

```diff
@@ -0,0 +1,115 @@
+use http;
+
+use proxy::http::h1;
+
+/// Describes the required HTTP "dialect".
+///
+/// This marks whether to use HTTP/2 or HTTP/1.x for a request. In
+/// the case of HTTP/1.x requests, it also stores a "host" key to ensure
+/// that each host receives its own connection.
+///
+/// TODO HTTP2 should indicate whether the requester supports PUSH_PROMISE.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum Dialect {
```

Comment:
TIOLI: perhaps calling this `http::Version` would be more descriptive? IDK...

Reply:
I avoided `Version` since that overloads the terminology from `http::Version` -- which is related but not really what we're trying to describe here: the Dialect is effectively a client configuration. We might construct two HTTP/1 stacks to a single endpoint that differ only in terms of whether we need absolute_uri normalization.  They share an `http::Version`, but are two distinct HTTP "dialects".

Reply:
Okay, fair enough!

Reply:
@briansmith That's more or less how I feel about it as well...